x-pack/plugins/search_playground/public/components/chat.tsx

```diff
@@ -82,11 +83,13 @@ export const Chat = () => {
     [messages]
   );
 
-  const regenerateMessages = () => {
+  const regenerateMessages = async () => {
+    setIsRegenerating(true);
```

Comment:
Why you didn't use `isLoading` from useChat? We probably want to disable those buttons on regular send messages too.

Reply:
`isLoading` only works for  when we `submit` the form. Thats why we had to handle this scenario separately.

Reply:
Aah got it. Should we consider to disable those buttons during submitting message too?